core/src/main/clojure/xtdb/types.clj

```diff
@@ -836,6 +852,29 @@
              (get-elem bb start typlen idx))
            (range length)))))
 
+(defn read-binary-text-array [^bytes ba]
+  (let [^ByteBuffer bb (ByteBuffer/wrap ba)
+        ;; _payload-size (.getInt bb 0)
+        ;; _dimensions (.getInt bb 4)
+        data-offset (.getInt bb 8)
+        ;; _oid (.getInt bb 12)
+        length (.getInt bb 16)
+        get-start (fn ^long [^long header-length ^long data-offset] (+ header-length data-offset))
+        start (get-start 20 data-offset)
+        start-of-str (fn ^long [^long start] (+ start 4))
+        next-start (fn ^long [^long start ^long str-len] (+ start 4 str-len))]
+    (:arr
```

Comment:
`loop` probably more in keeping with the rest of the codebase here, save creating a map on every iteration just to unwrap it on the next.